src/tests/api/Filter.test.js

```diff
@@ -0,0 +1,136 @@
+import { applyFilter } from 'src/api';
+import { filters } from './data-mocks/filters';
+
+const makePossibleFeature = (value) => [
+  {
+    type: 'Feature',
+    geometry: {
+      type: 'Point',
+      coordinates: [0, 0]
+    },
+    properties: {
+      column1: value
+    }
+  },
+  {
+    column1: value
+  }
+];
+
+describe('Filters', () => {
+  test('should return 1 if no filters present', () => {
+    const params = { filters: {}, type: 'number' };
+    expect(applyFilter(params)()).toBe(1);
```

Comment:
@AdriSolid shouldn't we use a feature for the test?

On the other hand, I find a bit strange the fact that applying a filter to an undefined feature return 'true'.... I think we should probably consider throwing an error in that case.

Reply:
- Was thinking that is not necessary passing a feature here, as long as if `filters` are equal to `{}`, the function firstly returns `1` or `true`: [reference](https://github.com/CartoDB/carto-react-lib/blob/a80ec3efca842b5c7bc683cc763277d316e86d27/src/api/Filter.js#L41)
- If we pass an undefined feature the result should be `0` or `false`, [because the feature doesn't pass the filter](https://github.com/CartoDB/carto-react-lib/blob/a80ec3efca842b5c7bc683cc763277d316e86d27/src/api/Filter.js#L23).

Reply:
Indeed, after reviewing a bit the code, I would suggest a refactor in `applyFilter` to `buildFeatureFilter` as the function doesn't apply the filter. It creates a function or 'filter' for you to apply later on a single feature. If this applyFilter is not exported in the public bundle, I would go for it

Reply:
- It is exported from the public bundle, and, I'm thinking that it is not necessary at all, going to remove it from there.
- Refactoring to `buildFeatureFilter`

Reply:
> * Was thinking that is not necessary passing a feature here, as long as `filters` are equal to `{}`, the function firstly returns `1` or `true`: [reference](https://github.com/CartoDB/carto-react-lib/blob/a80ec3efca842b5c7bc683cc763277d316e86d27/src/api/Filter.js#L41)
> * If we pass an undefined feature the result should be `0` or `false`, [because the feature doesn't pass the filter](https://github.com/CartoDB/carto-react-lib/blob/a80ec3efca842b5c7bc683cc763277d316e86d27/src/api/Filter.js#L23).

My point here is --> if we're testing buildFeatureFilter without filters and also without a feature, how do we know (from the test, without looking into function internals) that the result depends on the missing filter params and not the missing feature? Kind-of like we're having 2 variables involved not set